torch/onnx/_internal/exporter.py

```diff
@@ -681,21 +681,26 @@ def __init__(
         self._export_exception = export_exception
 
     def __call__(
-        self, *args: Any, options: Optional[ONNXRuntimeOptions] = None, **kwargs: Any
+        self,
+        *args: Any,
+        model: Union[torch.nn.Module, Callable, torch_export.ExportedProgram],
```

Comment:
I think we can design an api to load concrete weights for ONNXProgram (Concrefy - as opposed to fakefy). And make it lazy loading.

Both `__call__`, and `save` benefit from it.

Reply:
Yup, I believe we had discussed this not long ago.

That will be especially useful when fake mode is used to export super large models. the weight-by-weight loading from a checkpoint will be paramount to create an ONNX file with memory constraints.

I will create a dedicated PR for this; that might include a different format file if we cannot partially load conventional torch checkpoints

Reply:
I have two main concerns.

1. There is implicitly a "fake" version of `ONNXProgram`, where initializers, buffers etc are not filled in model_proto. This PR is on the direction of solidifying that concept. It is better to call it out explicitly and design apis around it.

2. I feel like the massive change of api is going backwards ... I feel it is mixing two use cases 1) fake export/ training where weights are not part of proto 2) aot export where constants are lifted and not part of proto. Is this a temporary change or do you plan to change it back in future dedicated PR?

Reply:
I don't remember why we are not filling in the lifted constants by aot_export into model_proto prior to this step. Could you elaborate?

Reply:
1. I didn't fully understand what you meant by "implicitly fake ONNXProgram". IIUC, `ONNXProgram` while in memory doesn't have weights (by design of fake tensor). But the `ONNXProgram.save` already fills the serialized ONNX file with initializers since when we implemented fake tensor support for the `torch._dynamo.export` fx tracer.

2. This PR does handle 1) fake tensor with "external state_dict" to call the exported model in memory (as the previous comment explained, `ONNXProgram.save` already was able to save a complete ONNX model with initializers) and 2) constants being lifted by AOTAutograd. 1) was the goal of the PR (fake tensor) and 2) ended up a side effect to enable a fake tensor test using google t5 or openai whisper (they had constants lifted). As we follow the resulting GraphModule as is and it has lifted constants, the resulting onnx model had the constant lifted. I have no strong preference over this; just made the smallest change to make it work

Reply:
> for 2. I saw the next PR changed it a bit, it does feel like we will be constantly changing it so this is more like temporary solution?

I am doing changes incrementally as the PRs are growing and the merge process is taking quite a bit to happen :)